notebooks/hive2bq/HiveToBigquery_parameterize_script.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from os import environ
+from typing import Dict, Sequence, Optional, Any
+import argparse
+
+import papermill as pm
+
+from parameterize_script import BaseParameterizeScript
+import parameterize_script.util.notebook_constants as constants
+from parameterize_script.util import get_common_args
+
+__all__ = ['HiveToBigQueryScript']
+
+
+class HiveToBigQueryScript(BaseParameterizeScript):
+
+    """
+    Script to parameterize HiveToBigQuery notebook.
+    """
+
+    @staticmethod
+    def parse_args(args: Optional[Sequence[str]] = None) -> Dict[str, Any]:
+        parser = argparse.ArgumentParser()
+
+        parser.add_argument(
+            f'--{constants.HIVE_METASTORE_ARG}',
+            dest=constants.HIVE_METASTORE,
+            required=True,
+            help='Hive metastore URI'
+        )
+
+        parser.add_argument(
+            f'--{constants.INPUT_HIVE_DATABASE_ARG}',
+            dest=constants.INPUT_HIVE_DATABASE,
+            required=True,
+            help='Hive database name'
+        )
+
+        parser.add_argument(
+            f'--{constants.INPUT_HIVE_TABLES_ARG}',
+            dest=constants.INPUT_HIVE_TABLES,
+            required=False,
+            default="*",
+            help='Comma separated list of Hive tables to be migrated "/table1,table2,.../" (Default: *)'
+        )
+
+        parser.add_argument(
+            f'--{constants.OUTPUT_BIGQUERY_DATASET_ARG}',
+            dest=constants.OUTPUT_BIGQUERY_DATASET,
+            required=True,
+            help='BigQuery dataset name'
+        )
+
+        parser.add_argument(
+            f'--{constants.TEMP_BUCKET_ARG}',
+            dest=constants.TEMP_BUCKET,
+            required=True,
+            help='GCS bucket name for temporary staging'
```

Comment:
I thought I saw a message from Surgit saying we should not use GCS and should use Cloud Storage instead. I suppose it doesn't matter too much if these scripts are for testing only but thought I'd mentioned it. Feel free to ignore if not relevant.

Reply:
Relevant catch, made changes wherever applicable